tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -1406,3 +1406,68 @@ def test_verify_attribute_of_users_are_updated(session, ldap_data, ldap_tear_dow
+
+
+@tier2
+def test_verify_ldap_filters_ipa(session, ipa_data, ldap_tear_down):
+    """Verifying ldap filters in authsource to restrict access
+
+    :id: 0052b272-08b1-11eb-80c6-0c7a158cbff4
+
+    :Steps:
+        1. Create authsource with onthefly enabled and ldap filter
+        2. Verify login from users according to the filter
+
+    :expectedresults: Login fails for restricted user
+
+    """
+    result = ssh.command(
+        cmd=f"echo {settings.ipa.password_ipa} | kinit admin", hostname=settings.ipa.hostname_ipa
+    )
+    assert result.return_code == 0
+    test_user = gen_string('alpha')
+    add_user_cmd = (
+        f"echo {settings.ipa.password_ipa} | ipa user-add {test_user} --first"
+        f"={test_user} --last={test_user} --password"
+    )
+    result = ssh.command(cmd=add_user_cmd, hostname=settings.ipa.hostname_ipa)
+    assert result.return_code == 0
+    ldap_auth_name = gen_string('alphanumeric')
+    with session:
+        session.ldapauthentication.create(
+            {
+                'ldap_server.name': ldap_auth_name,
+                'ldap_server.host': ipa_data['ldap_ipa_hostname'],
+                'ldap_server.server_type': LDAP_SERVER_TYPE['UI']['ipa'],
+                'account.account_name': ipa_data['ldap_ipa_user_name'],
+                'account.password': ipa_data['ldap_ipa_user_passwd'],
+                'account.base_dn': ipa_data['ipa_base_dn'],
+                'account.groups_base_dn': ipa_data['ipa_group_base_dn'],
+                'account.onthefly_register': True,
+                'attribute_mappings.login': LDAP_ATTR['login'],
+                'attribute_mappings.first_name': LDAP_ATTR['firstname'],
+                'attribute_mappings.last_name': LDAP_ATTR['surname'],
+                'attribute_mappings.mail': LDAP_ATTR['mail'],
+            }
+        )
+
+        # 'test_user' able to login before the filter is applied.
+        with Session(user=test_user, password=ipa_data['ldap_ipa_user_passwd']) as ldapsession:
+            with raises(NavigationTriesExceeded):
+                ldapsession.usergroup.search('')
+            assert test_user in ldapsession.task.read_all()['current_user']
+
+        # updating the authsource with filter
+        group_name = ipa_data['groups'].replace(" ", '').split(',')[0]
+        ldap_data = f"(memberOf=cn={group_name},{ipa_data['ipa_group_base_dn']})"
+        session.ldapauthentication.update(ldap_auth_name, {'account.ldap_filter': ldap_data})
+
+    # 'test_user' not able login as it gets filtered out
+    with Session(user=test_user, password=settings.ipa.password_ipa) as ldapsession:
+        with raises(NavigationTriesExceeded) as error:
+            ldapsession.user.search('')
+        assert error.typename == "NavigationTriesExceeded"
+
+    # delete the test_user from ipa
+    result = ssh.command(cmd=f"ipa user-del {test_user}", hostname=settings.ipa.hostname_ipa)
+    assert result.return_code == 0
```

Comment:
Looking at that, it seems user creation should be managed through fixture, so user is already there when test starts and is automatically removed after test finishes.
Right now, if any previous assertion fails for any reason, we will never run these lines and won't remove user. That might have impact on other tests, which assume that user does not exist.

Also, if concurrency is potentially an issue, please consider adding `@run_in_one_thread` decorator to test function.

Reply:
Yes, I will make it as a fixture. Thanks for the suggestion.
I have added `@run_in_one_thread` .